include/ozo/impl/result.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <ozo/error.h>
+#include <ozo/type_traits.h>
+#include <ozo/concept.h>
+#include <ozo/impl/row.h>
+
+namespace ozo::impl {
+
+template <typename Rows, typename OidMap, typename Iterator>
+Require<Iterable<Rows> && (ForwardIterator<Iterator> || InsertIterator<Iterator>), error_code>
+convert_result(Rows&& rows, const OidMap& oid_map, Iterator result) {
+    error_code ec = error::ok;
+    for (auto row : rows) {
+        if constexpr (ForwardIterator<Iterator>) {
+            ec = convert_row(std::move(row), oid_map, *result++);
+        } else if constexpr (InsertIterator<Iterator>) {
+            using row_value_t = typename Iterator::container_type::value_type;
+
+            auto result_row = row_value_t{};
```

Comment:
I think it is better to use an overloaded function instead of ```constexpr``` because even with ForwardIterator and not a default constructible type you still can get here a compile error here.